python/chronos/src/bigdl/chronos/autots/tspipeline.py

```diff
@@ -338,6 +353,126 @@ def load(file_path):
+               | TSDataset used in `AutoTSEstimator.fit`.
+               |
+               | 3. A torch.utils.data.dataloader.DataLoader object for calibration,
+               | Users should set the configs correctly (e.g. past_seq_len, ...).
+               | They can be found in TSPipeline._best_config.
+               |
+               | 4. A numpy ndarray tuple (x, y).
+               | x's shape is (num_samples, past_seq_len, input_feature_dim).
+               | y's shape is (num_samples, future_seq_len, output_feature_dim).
+               | They can be found in TSPipeline._best_config.
+
+        :param val_data: Same as calib_data,
+               should be data creator or TSDataset or DataLoader or numpy.ndarray.
+        :param metric: A str represent the metrics for tunning the quality of
+               quantization. You may choose from "mse", "mae", "rmse", "r2", "mape", "smape".
+        :param conf: A path to conf yaml file for quantization. Default to None,
+               using default config.
+        :param framework: string or list, [{'pytorch'|'pytorch_fx'|'pytorch_ipex'},
+               {'onnxrt_integerops'|'onnxrt_qlinearops'}]. Default: 'pytorch_fx'.
+               Consistent with Intel Neural Compressor.
+        :param approach: str, 'static' or 'dynamic'. Default to 'static'.
+        :param tuning_strategy: str, 'bayesian', 'basic', 'mse' or 'sigopt'. Default to 'bayesian'.
+        :param relative_drop: Float, tolerable ralative accuracy drop. Default to None,
+               e.g. set to 0.1 means that we accept a 10% increase in the metrics error.
+        :param absolute_drop: Float, tolerable ralative accuracy drop. Default to None,
+               e.g. set to 5 means that we can only accept metrics smaller than 5.
+        :param timeout: Tuning timeout (seconds). Default to 0, which means early stop.
+               Combine with max_trials field to decide when to exit.
+        :param max_trials: Max tune times. Default to 1. Combine with timeout field to
+               decide when to exit. "timeout=0, max_trials=1" means it will try quantization
+               only once and return satisfying best model.
+        """
+        from torch.utils.data import DataLoader, TensorDataset
+        from bigdl.chronos.data import TSDataset
+        # check model support for quantization
+        from bigdl.chronos.autots.utils import check_quantize_available
+        check_quantize_available(self._best_model.model)
+        # calib data should be set if the forecaster is just loaded
+        if calib_data is None and approach.startswith("static"):
+            raise RuntimeError("You must set a `calib_data` "
+                               "for quantization When you use 'static'.")
+        elif calib_data and approach.startswith("dynamic"):
+            raise RuntimeError("`calib_data` should be None When you use 'dynamic'.")
+
+        # change data tuple to dataloader
```

Comment:
line 423 - 431 and line 433 - 441 has nearly the same pattern, maybe we can use a function to clean the code.
```python
calib_data = preprocess_quantize_data(calib_data)
val_data = preprocess_quantize_data(val_data)
```